apiserver/gui.go

```diff
@@ -731,11 +730,16 @@ func (h *guiArchiveHandler) handlePost(w http.ResponseWriter, req *http.Request)
 	defer storage.Close()
 
 	// Read and validate the archive data.
-	data, hash, err := readAndHash(req.Body)
+	file, hash, cleanup, err := tmpCacheAndHash(req.Body)
 	if err != nil {
 		return errors.Trace(err)
 	}
-	size := int64(len(data))
+	defer cleanup()
+	fi, err := file.Stat()
```

Comment:
This would be simpler if we just made the tmpCacheAndHash() method return the hash *and* the size?
eg
`file, hash, size, cleanup, err := tmpCacheAndStats(req.Body)`